packages/now-routing-utils/src/index.ts

```diff
@@ -101,63 +84,77 @@ export function normalizeRoutes(
             null,
             2
           )}`,
-          errors
+          errors,
         }
       : null;
 
   return { routes, error };
 }
 
-/**
- * An ajv schema for the routes array
- */
-export const schema = {
-  type: 'array',
-  maxItems: 1024,
-  items: {
-    type: 'object',
-    additionalProperties: false,
-    properties: {
-      src: {
-        type: 'string',
-        maxLength: 4096
-      },
-      dest: {
-        type: 'string',
-        maxLength: 4096
-      },
-      methods: {
-        type: 'array',
-        maxItems: 10,
-        items: {
-          type: 'string',
-          maxLength: 32
-        }
-      },
-      headers: {
-        type: 'object',
-        additionalProperties: false,
-        minProperties: 1,
-        maxProperties: 100,
-        patternProperties: {
-          '^.{1,256}$': {
-            type: 'string',
-            maxLength: 4096
-          }
-        }
-      },
-      handle: {
-        type: 'string',
-        maxLength: 32
-      },
-      continue: {
-        type: 'boolean'
-      },
-      status: {
-        type: 'integer',
-        minimum: 100,
-        maximum: 999
-      }
+export function getTransformedRoutes({
+  nowConfig,
+  filePaths,
+}: GetRoutesProps): NormalizedRoutes {
+  const { cleanUrls, rewrites, redirects, headers, trailingSlash } = nowConfig;
+  let { routes } = nowConfig;
+  const errors: { message: string }[] = [];
+  if (typeof routes !== 'undefined') {
+    if (typeof cleanUrls !== 'undefined') {
+      errors.push({
+        message: 'Cannot define both `routes` and `cleanUrls`',
+      });
+    }
+    if (typeof redirects !== 'undefined') {
+      errors.push({
+        message: 'Cannot define both `routes` and `redirects`',
+      });
+    }
+    if (typeof rewrites !== 'undefined') {
+      errors.push({
+        message: 'Cannot define both `routes` and `rewrites`',
+      });
+    }
+    if (typeof headers !== 'undefined') {
+      errors.push({
+        message: 'Cannot define both `routes` and `headers`',
+      });
+    }
+    if (typeof trailingSlash !== 'undefined') {
+      errors.push({
+        message: 'Cannot define both `routes` and `trailingSlash`',
+      });
+    }
+  } else {
+    routes = [];
+    if (typeof cleanUrls !== 'undefined') {
+      routes.push(...convertCleanUrls(filePaths));
+    }
+    if (typeof redirects !== 'undefined') {
+      routes.push(...convertRedirects(redirects));
+    }
+    if (typeof rewrites !== 'undefined') {
+      routes.push(...convertRewrites(rewrites));
+    }
+    if (typeof headers !== 'undefined') {
+      routes.push(...convertHeaders(headers));
```

Comment:
I think we should move `headers` above `rewrites`

Reply:
I moved headers above rewrites, thanks!